sphinx_doc/tutorials/docs/writing_new_costmap2d_plugin.rst

```diff
@@ -0,0 +1,195 @@
+.. _writing_new_costmap2d_plugin:
+
+Writing a new Costmap2D plugin
+******************************
+
+- `Overview`_
+- `Requirements`_
+- `Tutorial Steps`_
+
+Overview
+========
+
+This tutorial shows how to create your own simple `ROS plugin <http://wiki.ros.org/pluginlib>`_ for Costmap2D.
+
+Before starting the tutorial, please check this `video <https://vimeo.com/106994708>`_ which contains information about Costmap2D layers design and plugins basic operational principals. Also, it is highly recommended to proceed :ref:`stvl` tutorial to get understanding about loading and using external plugin mechanisms.
+
+Requirements
+============
+
+- Ensure ROS2 and Gazebo are installed or built locally. For more details please refer for ROS2 installation web-page: https://index.ros.org/doc/ros2/Installation/.
+- You must have Navigation2 project built locally. To proceed it please follow :ref:`master-build` instructions.
+- Turtlebot3 packages should be installed from apt repositories or built locally:
+
+.. code-block:: bash
+
+  $ mkdir -p ~/turtlebot3_ws/src
+  $ cd ~/turtlebot3_ws
+  $ wget https://raw.githubusercontent.com/ROBOTIS-GIT/turtlebot3/ros2/turtlebot3.repos
+  $ vcs import src < turtlebot3.repos
+  $ colcon build --symlink-install
+
+Then add the following lines to ``~/.bashrc``
+
+.. code-block:: bash
+
+  source ${HOME}/turtlebot3_ws/install/setup.bash
+  export ROS_DOMAIN_ID=30
+  export TB3_MODEL=waffle
+  export TURTLEBOT3_MODEL=${TB3_MODEL}
+  export GAZEBO_MODEL_PATH=$GAZEBO_MODEL_PATH:${HOME}/turtlebot3_ws/src/turtlebot3/turtlebot3_simulations/turtlebot3_gazebo/models
+
+and source it:
+
+.. code-block:: bash
+
+  $ source ~/.bashrc
+
+Tutorial Steps
+==============
+
+1- Write a new Costmap2D plugin
+-------------------------------
+
+For better demonstration an example plugin will put repeating costs gradients to costmap. Therefore plugin's name was chosen as ``GradientLayer``. It is simply based on ``InflationLayer`` without unnecessary for tutorial purposes functionality.
```

Comment:
I don't think you need to bring up the inflation layer, lets just stick on topic of this toy example for creating a costmap plugin